engine/classes/ElggEntity.php

```diff
@@ -1686,7 +1686,14 @@ protected function update() {
 		unset($persisted_entity);
 
 		if ($allow_edit) {
-			$allow_edit = _elgg_services()->events->trigger('update', $this->type, $this);
+			$allow_edit = elgg_trigger_before_event('update', $this->type, $this);
```

Comment:
Technically all this needs to work is the after event.  I could leave the old update event in place, it just seemed logical that if I was adding the after event it should have a matching before event